pkg/tfbridge/info.go

```diff
@@ -111,6 +114,31 @@ type ProviderInfo struct {
 	SkipExamples func(SkipExamplesArgs) bool
 }
 
+// DocRuleInfo controls file discovery and edits for any subset of docs in a provider.
+type DocRuleInfo struct {
+	// A list of edit rules to apply.
+	EditRules []DocsEdit
+
+	// A function to suggest alternative file names for a TF element.
+	AlternativeNames func(DocsPathInfo) []string
+}
+
+// Information for file lookup.
+type DocsPathInfo struct {
```

Comment:
Curious, how will this struct grow? Adding DataSource id so AlternativeNames can find datasource?

Reply:
I'm not sure. We might add a `TokenKind: Enum(Resource|DataSource)` indicator. I can also see the benefit of adding a `DocsPath: string` field.

Given how expensive these kinds of changes are, I though making it a struct preemptively would save us pain later.

Reply:
Just call it token and mention its resource or datasource token in the comment? I think that' what it is, but internal variables say "resource" de facto it's R or D.